test/00_operators/ReductionTests.cu

```diff
@@ -225,6 +225,44 @@ TYPED_TEST(ReductionTestsNumericNoHalf, Sum)
       }
     }
   }
+  
+  {
+    tensor_t<TypeParam, 2> t2a({30, 40});
+    tensor_t<TypeParam, 2> t2b({30, 40});
+    
+    auto t4 = ones<TypeParam>({30, 40, 50, 60});
+
+    sum(t2a, t4, 0);
+    sum(t2b, t4, {2,3}, 0);
+
+    cudaStreamSynchronize(0);
+    for (index_t i = 0; i < t2a.Size(0); i++) {
+      for (index_t j = 0; j < t2a.Size(1); j++) {
+        EXPECT_TRUE(MatXUtils::MatXTypeCompare(
+            t2a(i, j), t2b(i,j)));
+      }
+    }
+  }
+
+  
+  {
+    tensor_t<TypeParam, 2> t2a({50, 60});
+    tensor_t<TypeParam, 2> t2b({50, 60});
+    
+    auto t4 = ones<TypeParam>({30, 40, 50, 60});
+
+    sum(t2a, permute(t4,{2,3,0,1}), 0);
+    sum(t2b, t4, {0,1}, 0);
+
+    cudaStreamSynchronize(0);
+    for (index_t i = 0; i < t2a.Size(0); i++) {
+      for (index_t j = 0; j < t2a.Size(1); j++) {
+        EXPECT_TRUE(MatXUtils::MatXTypeCompare(
+            t2a(i, j), t2b(i,j)));
```

Comment:
Have you verified these match exactly with what Python does? It would be useful to have all reduction tests match Python, but that's a bigger task.

Reply:
yes